api/v1beta1/disruption_webhook.go

```diff
@@ -100,6 +104,17 @@ func (r *Disruption) ValidateCreate() error {
 		return fmt.Errorf("invalid disruption name: %w", err)
 	}
 
+	if safemodeEnvironment != "" {
+		disruptionEnv, ok := r.Annotations[safeModeEnvironmentAnnotation]
+		if !ok {
+			return fmt.Errorf("disruption does not specify an environment to run, but this controller requires it. Set the annotation `%s:\"%s\"` to run on this controller", safeModeEnvironmentAnnotation, safemodeEnvironment)
+		}
+
+		if disruptionEnv != safemodeEnvironment {
+			return fmt.Errorf("disruption is configured to run in %s but has been applied in %s", disruptionEnv, safemodeEnvironment)
```

Comment:
Can we specify where the value comes from in that error log too please (like the one above) just in case it's a mystery to the one applying the disruption?